### PR TITLE
Segments

### PR DESCRIPTION
**Pull request by laliconfigcat**
### Describe the purpose of your pull request

Added Segments as a data source and resource too. 
Added Segments to the setting_value resource.

### Requirement checklist (only if applicable)

- [x] I have covered the applied changes with automated tests.
- [x] I have executed the full automated test set against my changes.
- [x] I have validated my changes against all supported platform versions.
- [ ] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: docs/resources/segment.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# configcat_segment Resource
+
+Creates and manages a **Segment**.
```
Comment (sigewuzhere) [inline docs/resources/segment.md:3]:
```suggestion
Creates and manages a **Segment**. Read more about segmentation and segments in ConfigCat [here].
```

File: docs/data-sources/segments.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# configcat_segments Resource
+
+Use this data source to access information about existing **Segments**.
```
Comment (sigewuzhere) [inline docs/data-sources/segments.md:3]:
```suggestion
Use this data source to access information about existing **Segments**. Read more about segmentation and segments in ConfigCat [here].
```

File: docs/resources/setting_value.md [hunk 3]
```diff
@@ -73,10 +73,13 @@ The Feature Flag/Setting's value
...
+Either the comparison_attribute/comparator/comparison_value attributes or the segment_comparator/segment_id attributes are required together.
 
-* `comparison_attribute` - (Required) The [comparison attribute](https://configcat.com/docs/advanced/targeting/#attribute).
-* `comparator` - (Required) The [comparator](https://configcat.com/docs/advanced/targeting/#comparator).
-* `comparison_value` - (Required) The [comparison value](https://configcat.com/docs/advanced/targeting/#comparison-value).
+* `comparison_attribute` - (Optional) The [comparison attribute](https://configcat.com/docs/advanced/targeting/#attribute).
+* `comparator` - (Optional) The [comparator](https://configcat.com/docs/advanced/targeting/#comparator).
+* `comparison_value` - (Optional) The [comparison value](https://configcat.com/docs/advanced/targeting/#comparison-value).
+* `segment_comparator` - (Optional) The segment_comparator. Possible values: isIn, isNotIn.
+* `segment_id` - (Optional) The Segment's unique identifier.
```
Comment (sigewuzhere) [inline docs/resources/setting_value.md:82]:
```suggestion
* `segment_id` - (Optional) The [Segment](link to CC docs segments)'s unique identifier.
```

File: docs/guides/segments.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
+# Advanced usage of Resources with Segments
+
```
Comment (sigewuzhere) [inline docs/guides/segments.md:2]:
```suggestion
Read more about segmentation and segments in ConfigCat [here].
```

**Review by sigewuzhere — CHANGES_REQUESTED** [review]

File: docs/resources/segment.md (hunk 1 above)
Comment (laliconfigcat) [reply docs/resources/segment.md]: A többihez hasonlóan "What is a Segment.." lett a leírás.

File: docs/data-sources/segments.md (hunk 2 above)
Comment (laliconfigcat) [reply docs/data-sources/segments.md]: A többihez hasonlóan "What is a Segment.." lett a leírás.

File: docs/guides/segments.md (hunk 4 above)
Comment (laliconfigcat) [reply docs/guides/segments.md]: Done

**Review by sigewuzhere — APPROVED** [review]

**Pull request merged**